components/form/__tests__/index.test.tsx

```diff
@@ -1778,6 +1779,48 @@ describe('Form', () => {
     expect(container.querySelector('.ant-form-item-has-error')).toBeTruthy();
   });
 
+  it('custom feedback icons should display when pass hasFeedback prop', async () => {
+    const App = ({ trigger = false }: { trigger?: boolean }) => {
+      const form = useRef<FormInstance<any>>(null);
+
+      useEffect(() => {
+        if (!trigger) return;
+        form.current?.validateFields();
+      }, [trigger]);
+
+      return (
+        <Form ref={form}>
+          <Form.Item
+            label="Success"
+            name="name1"
+            hasFeedback={{
+              icons: () => ({
+                error: <AlertFilled id="custom-error-icon" />,
+              }),
+            }}
```

Comment:
I have more thoughts about usage now:

```jsx
<Form>
  <Form.Item hasFeedback={{ icons: ... }} />
  <Form.Item hasFeedback={{ icons: ... }} />
  <Form.Item hasFeedback={{ icons: ... }} />
</Form>
```

vs

```jsx
<Form feedbackIcons={{ ... }}>
  <Form.Item hasFeedback />
  <Form.Item hasFeedback />
  <Form.Item hasFeedback />
</Form>
```

Which is better for real world?

Reply:
@afc163  Need both. For example,

if user use `Form.Item` independently (just use the design), `Form.Item` should have `hasFeedback`.

if user use `Form.Item` with `Form`, `hasFeedback` better on `Form`.

Reply:
Both +1. But secondary is necessary.

Reply:
The first should be `<Form hasFeedback={{ icons: {} }} >`, props in Form and Form.Item are both needed for configuration separately or together.

Reply:
Both